packages/cxl-ui/src/components/cxl-base-card.js

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import { LitElement, html } from 'lit';
+import { property, state } from 'lit/decorators.js';
+import '@vaadin/details';
+import '@vaadin/button';
+import cxlBaseCardStyles from '../styles/cxl-base-card-css.js';
+
+export class CXLBaseCardElement extends LitElement {
+  static get styles() {
+    return [cxlBaseCardStyles];
+  }
+
+  separator = html`<span> | </span>`;
+
+  @state() _tagsHasChildren = false;
+
+  @property({ type: String }) id = '';
+
+  @property({ type: String }) theme = 'Base';
+
+  @property({ type: String }) name = '';
+
+  @property({ type: String }) time = '';
+
+  @property({ type: String }) instructor = '';
+
+  @property({ type: String }) avatar = '';
+
+  @property({ type: Boolean, reflect: true }) new = false;
+
+  _slotHasChildren (e) {
+    const slot = e.target
+    const { name } = slot
+    const children = slot.assignedNodes()
+    this[`_${name}HasChildren`] = !!children.length
+  }
+
+  _renderHeader() {
+    return html`
+      <header>
+        <div class="info">
+          <div class="tags">
+            ${this.theme ? html`<span class="tag">${this.theme}</span>`: ''}
+            ${this.theme && this._tagsHasChildren ? this.separator : ''}
+            <slot name="tags" @slotchange=${this._slotHasChildren}></slot>
+            ${this.new ? html`${this.theme ? this.separator : ''}<span class="tag new">NEW</span>` : '' }
+          </div>
```

Comment:
As I understand, light card will need to get rid of this part, which means copy-pasting this method anyway, just without `.tags`.

What do you think about one of these solutions:
1. Add `usesTags` boolean flag, make it `false` for light card and render `.tags` part conditionally.
2. Compose `_renderHeader()` with other methods:
    * `_renderHeaderTags()`
    * `_renderHeaderName()`
    * `_renderHeaderAttributes()`
    * ...
and then skip unnecessary methods in child classes.

Reply:
I would just use the boolean to match CSS to hide the tags div. Having several templates just to be able to cover that one case feels like overkill. The css would be pretty simple:
```
:host(:not([usesTags]) {
    header .tags { display: none; }
}
```
That can be used for other parts of the template too if, in the future, we need more card layout variations.

Reply:
My idea was to render this part conditionally in `_renderHeader()`, not to hide it with CSS. Not much an overkill

```js
<div class="tags">
  ${this.theme ? html`<span class="tag">${this.theme}</span>`: ''}
  ${this.theme && this._tagsHasChildren ? this.separator : ''}
  <slot name="tags" @slotchange=${this._slotHasChildren}></slot>
  ${this.new ? html`${this.theme ? this.separator : ''}<span class="tag new">NEW</span>` : '' }
</div>
```